src/gs-shell.c

```diff
@@ -924,6 +924,13 @@ side_filter_row_selected (GtkListBox *side_filter,
 	}
 }
 
+static gboolean
+side_filter_row_has_category (GsSideFilterRow *row,
+			      GsCategory *category)
+{
+	return gs_side_filter_row_get_category (row) == category;
```

Comment:
Sorry to nitpick, but could you move this check directly in the loop, without a separate function?